R/s3.R

```diff
@@ -0,0 +1,163 @@
+##' S3 backend for storr
+##'
+##' @title S3 backend for rds object cache driver
+##'
+##' @param bucket Name of the S3 bucket for which you wish to create
+##'   of connect store
+##'
+##' @param remote_root Base path on the remote bucket
+##'
+##' @param ... Additional arguments passed through to
+##'   \code{storr::driver_rds}
+##'
+##' @param path_local Optional path to a local cache (see
+##'   \code{storr::driver_remote})
+##'
+##' @param default_namespace Default namespace (see \code{storr::storr})
+##' @export
+storr_s3 <- function(bucket, remote_root, ..., path_local = NULL,
+                     default_namespace = "objects") {
+  dr <- driver_s3(bucket, remote_root, ..., path_local = path_local)
+  storr::storr(dr, default_namespace)
+}
+
+
+##' @export
+##' @rdname storr_s3
+driver_s3 <- function(bucket, remote_root, ..., path_local = NULL) {
+  ops <- s3_file_ops(bucket, remote_root)
+  storr::driver_remote(ops, ..., path_local = path_local)
+}
+
+
+s3_file_ops <- function(bucket, root) {
+  R6_s3_file_ops$new(bucket, root)
+}
+
+
+R6_s3_file_ops <- R6::R6Class(
+  "s3_file_ops",
+
+  public = list(
+    bucket = NULL,
+
+    initialize = function(bucket, root) {
+      self$bucket <- bucket
+      self$root <- root
+    },
+
+    type = function() {
+      "s3"
+    },
+
+    destroy = function() {
+      ## TODO: not sure if this is right?
```

Comment:
This looks correct to me. Should delete the entire store from the root. Will test when I get it working with PR#74 on storr.